src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/Configuration/RabbitMqHostConfigurator.cs

```diff
@@ -151,5 +151,10 @@ string GetVirtualHost(Uri address)
 
             throw new FormatException("The host path must be empty or contain a single virtual host name");
         }
+
+        public void ConnectionName(string connectionName)
+        {
+            _settings.ClientProvidedName = connectionName;
```

Comment:
Note that this allows setting the connection name to an empty string `""`, where the `RabbitMqHostConfigurator` constructor explicitly ignores empty strings:

https://github.com/MassTransit/MassTransit/blob/7f8220b9b5d805b301941b06805f06e37e43c447/src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/Configuration/RabbitMqHostConfigurator.cs#L25-L26

I have tested the connection when passing an empty string as name, it doesn't seem to cause any issues. In my opinion, adding an `IsNullOrEmpty(...)` check in this setter function creates unexpected behavior (ie. the user sets a value but we silently ignore it), so I have left it out.